### DISCUSSION TITLE
不便だと思うことを書く場所

### DISCUSSION THREAD

**Discussion by Kai-Z-JP** [Kai-Z-JP/KaizPatchX#191] [Opinion]
多分不便なことがあると思うので

**Comment by Trollebas** [comment]
Hi Kais, I will now describe what's inconvenient in the mod on my behalf - a person who knows RTM for 5 years, who created contact network packs and has extensive experience solving problems with RTM in the Russian player community.
First, we need to rewrite train initialization, so they don't require chunk loading at all times, so trains don't disappear and turn into empty buggy pink creatures with questions.
Secondly, rewrite the rendering of wires and insulators, as the current system creates lags. It would be desirable to add a separate wire type for the contact line - so that no traffic light signal is conducted through it.
Third, redesign the train selection system - either a "confirm" button to change colour, train name or other properties, or make it so the player can guess to save the settings and see the changes when the train is saved.
Fourth, improve the crafting system, as today RTM in survival is something unknown and incomprehensible.
Fifth, make an autocoupler system so that trains are coupled automatically without switching to carriage coupling mode. Perhaps there are coupler types that require separate preparation before coupling, for them add a parameter that returns this type of coupling for a given train.
Sixth, add possibility of getting on train as driver not by click on cart, but by coordinates from JSON train, i.e. on seat or in place prescribed by coordinates.
Seventh, remove the limit on the number of traffic lights, this would allow you to create different types of traffic lights.
Eighth, it would be nice to see your current system for barriers and other elements on the red light signal, using rewritten and non-lagging RTM wires.
Tenth, solve the problem of missing rails, for example by creating straight sections of 16 blocks each.
Tenth, finally implement middle button copying of diagonal marker rails.

Rus:
Привет, Кайз, сейчас я буду расписывать то, что неудобно в моде от своего лица – человека, знающего RTM уже 5 лет, создававшего паки на контактную сеть и имеющего огромный опыт решения проблем с RTM в русском сообществе игроков.
Во-первых, нужно переписать инициализацию поездов, чтобы им не требовалась загрузка чанков всегда, чтобы не происходило пропадание поездов и их превращения в пустые ошибочные розовые существа с вопросами.
Во-вторых, требуется переписать визуализацию проводов и изоляторов, так как нынешняя система создаёт лаги. Желательно добавить отдельный тип провода для контактной сети – чтобы через него не проводился сигнал светофора.
В-третьих, переработать систему выбора поездов – нужно добавить либо кнопку «подтвердить» для изменений цвета, названия поезда или других характеристик, либо сделать это так, чтобы игрок мог догадаться эти настройки сохранить и увидеть изменения при спавне поезда.
В-четвёртых, улучшить систему крафтов, так как сегодня RTM в выживании – это что-то неизвестное и непонятное.
В-пятых, сделать систему автосцепки, чтобы поезда сцеплялись автоматически без перехода в режим сцепления вагонов. Возможно, есть типы сцепок, которые требуют их отдельной подготовки перед сцеплением, для них добавить параметр, возвращающий для данного поезда такой вид сцепки.
В-шестых, добавить возможность посадки в поезд в качестве машиниста не кликом по тележке, а по координатам из JSON поезда, то есть, на кресло или в место, прописанное по координатам.
В-седьмых, убрать ограничение на количество сигналов светофора, это позволит создавать разные типы светофоров.
В-восьмых, было бы неплохо увидеть свою систему тока для шлагбаумов и других элементов на сигнале красного камня с применением переписанных и не лагающих проводов RTM.
В-девятых, решить проблему пропадающих рельс, например, созданием прямых участков по 16 блоков.
В-десятых, наконец, реализовать копирование по средней кнопке мыши рельс диагонального маркера.

**Comment by Trollebas** [comment]
Предлагаю кликом средней мыши по LinePole или изолятору копировать его изменённое положение.
I suggest that a middle mouse click on a LinePole or insulator copies its changed position.

**Reply by Kai-Z-JP** [reply]
I implemented it in #221. Thank you for your idea.

**Comment by Saqliva** [comment]
モデルの更新とかデバッグとかでいちいちマイクラを再起動するのがやってられないのでゲーム内でモデルとかjsonを再読み込みする機能が欲しい

**Reply by Kai-Z-JP** [reply]
確かに面白そうやってみる

**Comment by Saqliva** [comment]
スクリプトの文法ミスとかでクラッシュせずにパックが読み込まれなくなるせいでパックのバグの原因を探すのに一苦労するのでスクリプトのミスとかでクラッシュするとかログ吐くとかのデバッグモードがほしい
切り替えはconfigとか

**Reply by Kai-Z-JP** [reply]
修正@#275

**Comment by Builder256** [comment]
描画スクリプトの文法ミスがあった際にクラッシュしないのでクラッシュするようにしてほしいです

**Reply by Kai-Z-JP** [reply]
修正@#275

**Comment by dannbouru** [comment]
NPCの設定を保存できるようにして欲しいです。

**Reply by Kai-Z-JP** [reply]
NPCの設定を保存の意味がよくわからないので詳しく説明してもらえると助かります

**Reply by dannbouru** [reply]
NPCを大量に電車に乗せる時に、一人ずつSUICAを持たせたり、passenger状態にすることが非常に不便なので、アイテム電車や照明などと同じような感じで設定できるようにして欲しいです。

**Comment by dbsanrtm** [comment]
rtmdengoadapterを使用して運転する際、
車掌が車掌弁を扱っても非常ブレーキが運転士側のマスコン入力に上書きされてしまい
非常停止が不可能なので非常停止ができるように上書きされないようにしてほしいです。

**Reply by Kai-Z-JP** [reply]
なんとかします

**Comment by Saqliva** [comment]
#303 と#311 で変更された踏切の可聴範囲がとんでもなく広すぎる
地下に存在する踏切モデルの音声が地上にいても聞こえるからどうにかならないかな
※サウンド設定で調節してもなかなかうるさい

**Reply by Kai-Z-JP** [reply]
#346 で再度修正済み

**Reply by Saqliva** [reply]
まだ現行バージョンに実装されてなかったから気づかなかった
申し訳ない

**Comment by MuffinOooo** [comment]
走行音を再生する距離が車両から20~30mくらいまで聞こえるようにして欲しいです。

**Reply by Kai-Z-JP** [reply]
Configから変更可能です(In-Gameでも)

**Reply by Kai-Z-JP** [reply]
(1.7.0-beta.1以降)

**Reply by MuffinOooo** [reply]
そうなのですね、ありがとうございます。

**Comment by FujiwaraAzusa** [comment]
昔から長らくKaizPatchXのお世話になっております。久々にjarファイルダウンロードしようと思ってgithub見たら12時間前に更新されてて驚きました。直線用マーカーいいですね。

以下要望です

・従来からのRTM専用作業台のバグが多いのと、多くのRTMアイテムやブロックがクラフト非対応だったりする点
・サバイバルで専用作業台や信号変換機、移動装置を破壊するとアイテムとしてドロップしなかったり、別のブロックになってしまう点
・蛍光灯や階段、券売機などのブロックの当たり判定的や、カーソルを合わせたときの当たり判定の表示される枠が大きいかったり見栄えが悪くなる点
・碍子や改札機、装飾系のブロック等の読み込み時に転換機のモデルが最初に表示される点（根幹のシステム上不可能だったら申し訳ないです）

これら可能でしたら修正して欲しいです。注文多くてすみません()

**Reply by Kai-Z-JP** [reply]
1, 2: 基本的にKaizPatchは特に要請が無いためサバイバルで動作確認をしておらず、クラフトなどはバグ修正していません。 使用者の数的にも積極的なバグ修正は行うことは考えていないため、そのような事象は個別にIssueを立てて頂けると助かります
3: 当たり判定は1.12.2の車両の当たり判定が見えるようになる機能を移植しようとしたのですが失敗しており (#230)、ブロックへの実装は尚更難しいかつ運転に支障が無いため優先度は低めです
4: 86bcd4e4cfcd689c5e75db83586d861f6f9fb94c で修正しました

**Reply by FujiwaraAzusa** [reply]
了解しました

**Comment by kanamedango** [comment]
警笛や進段などのキー入力を取得できる関数が欲しいです。
警笛キーを押す時間によって電笛か空笛か切り替えられる機能を実装したいと考えています。
